### PR TITLE
Issue 3843: Refactor DRM message handler

### PR DESCRIPTION
**Pull request by mwatson2**
Simplify processing path:
1. message mapping (keysystem-specific)
2. message handler
3. request constructor (server-specific)

Pass generic `params` variable to enable easier addition of new params.

### CODE REVIEW

**Comment by jdsmith3000** [comment]
This LGTM.  Please hold merge for review by a AlexeyKor.

File: encrypted-media/util/drm-messagehandler.js [hunk 1]
```diff
@@ -37,229 +38,171 @@ drmconfig = {
...
-                    delete headers.Content;
-                }
-                return headers;
+    // request = { url: ..., headers: ..., body: ... }
+    //
+    // content = { assetId: ..., variantId: ..., key: ... }
+    // params = { expiration: ... }
+
+    'drmtoday': function(config, sessionType, content, messageType, message, headers, params) {
+        var optData = JSON.stringify( { merchant: config.merchant } );
```
Comment (AlexeyKor) [inline encrypted-media/util/drm-messagehandler.js]: nit: spacing inconsistency. (103, 105, 108, 110, 117 lines)

File: encrypted-media/util/drm-messagehandler.js [hunk 2]
```diff
@@ -37,229 +38,171 @@ drmconfig = {
...
         }
-    };
 
-    return new Promise(function(resolve, reject) {
-        var keysystemfns = keySystems[this._keysystem],
-            serverfns,
-            url = undefined,
-            requestheaders = {},
-            credentials = undefined;
+        return JWT.encode( "HS256", { optData: optData, crt: JSON.stringify([crt]) }, config.secret ).then(function(jwt){
```
Comment (AlexeyKor) [inline encrypted-media/util/drm-messagehandler.js]: nit: spacing inconsistency (122, 131, 133, 136)

File: encrypted-media/util/drm-messagehandler.js [hunk 3]
```diff
@@ -37,229 +38,171 @@ drmconfig = {
...
+}
 
-        serverfns.getCustomHeaders( this._drmconfig, this._sessionType, messageType, this._content, expiration, variantId ).then(function(customHeaders) {
+MessageHandler.prototype.messagehandler = function messagehandler(messageType, message, responseType, headers, params) {
 
-            updateHeaders( customHeaders );
-            updateHeaders( keysystemfns.getRequestHeadersFromMessage(message) );
+    var variantId = params ? params.variantId : undefined;
+    var key;
+    if( variantId ) {
```
Comment (AlexeyKor) [inline encrypted-media/util/drm-messagehandler.js]: nit: spacing inconsistency

**Review by AlexeyKor — APPROVED** [review]
Looks good to me.

**Pull request merged**